all: update WASI 0.2 bindings

ydnar opened the pull request:
- Update to `go.bytecodealliance.org@v0.6.2`
- Update package `cm` to `go.bytecodealliance.org/cm@v0.2.2`
- Regenerate WASI 0.2 bindings in `internal/wasi`

Full changelog: https://github.com/bytecodealliance/go-modules/blob/main/CHANGELOG.md

ydnar committed:
GNUmakefile, internal/wasm-tools: update to go.bytecodealliance.org@v0.6.0

ydnar committed:
internal/cm: update to go.bytecodealliance.org/cm@v0.2.0

ydnar committed:
internal/wasi: regenerate WASI 0.2 bindings

Use go.bytecodealliance.org/cmd/wit-bindgen-go@v0.6.0

ydnar committed:
internal/cm: exclude certain files when copying package

ydnar committed:
internal/cm: remove go.mod file

b0ch3nski commented:
Somehow the regeneration adds an executable bit on the files which is not needed I think?

ydnar committed:
internal/wasm-tools, internal/cm: udpate to go.bytecodealliance.org@v0.6.1 and cm@v0.2.1

ydnar commented:
> Somehow the regeneration adds an executable bit on the files which is not needed I think?

Yes, odd. Let me look.

ydnar commented:
Thanks! Fixed here: https://github.com/bytecodealliance/go-modules/pull/316

ydnar committed:
internal/wasi: remove x bit from generated files

ydnar committed:
internal/cm: change error type from struct{string}

This is a temporary fix until cm@v0.2.2 is released, and package internal/cm and internal/wasi can be regenerated.

See #4810 for more information.

ydnar committed:
internal/wasm-tools: update to go.bytecodealliance.org@v0.6.2

deadprogram commented:
@ydnar perhaps you want to rebase this against the latest dev for testing?

deadprogram commented:
Hard for me to evaluate this PR except for "the tests are passing". From that point of view, we can merge now.

ydnar commented:
Key changes include:

1. Migration to new Go package path : go.bytecodealliance.org
2. Removed unnecessary generated .wit file in internal/wasi
3. Bug fixes and improvements

Full changelog: https://github.com/bytecodealliance/go-modules/blob/main/CHANGELOG.md

deadprogram commented:
Now merging, thank you!
